### ISSUE TITLE
How to integrate Handlebars 

### ISSUE THREAD

**Issue by ladvishal1985** [zalando/tailor#241]
I have a requirement where I want to have few parts application to be handled through tailor and rest through templating engine like Handlebars. I believe using any templating engine will help us in managing views more efficiently. How easy it would be to use Handlebars with the current tailor.js setup ?

**Comment by vigneshshanmugam** [comment]
Yes it should be possible if you do the template compilation in a fragment server

### Example 1
```js
<div id="template">
     <fragment src="http://handle-bar-fragment.blah"/> 
</div>
```

In your fragment server.js, you can do
```js
var http = require("http");
var Handlebars = require("handlebars");
var compiled = Handlebars.precompile(template);

http.createServer((req, res) => {
   var template = "<div>whatever {{ name }}</div>";
   var compiled = Handlebars.precompile(template);
   // stream the template code
   response.end(compiled);
})
```


### Example 2

If you don't want to run servers, you can simply use the handlebar runtime like you do in regular HTML

```html
<script src="/libs/handlebars.runtime.js"></script>
 
<script id="entry-template" type="text/x-handlebars-template">
  <div class="entry">
    <h1>{{title}}</h1>
    <div class="body">
      {{body}}
    </div>
  </div>
</script> 

<script>
var source   = document.getElementById("entry-template").innerHTML;
var template = Handlebars.compile(source);
</script>
```

You can also have fragments in the above HTML and tailor would stream them like regular setup.

**Comment by ladvishal1985** [comment]
This helps. I want to set Handlebar as the view engine in the tailor app and not in the fragment. Actually I was also trying to figure how how can I dynamically set variables in the HTML of Tailor app. For example I see specify <fragment src="localhost:3000/contact" > How is it possible to set localhost via config variable. I might sound little dumb but I am unable to figure out this as we are not using any view engine within the tailor app.

**Comment by vigneshshanmugam** [comment]
Can you post an example html of how you would like to get it working?

**Comment by ladvishal1985** [comment]
Let say I have base layout defined this way:
<!doctype html>
<html>
    <head>
        <title>Sample </title>
       <script src="//code.jquery.com/jquery-2.0.2.min.js"></script>
    </head>
    <body>
        <header><img src='/img/logo.png' alt='Logo'></header>
        {{{body}}}
    </body>
</html>

Child views defined like this(I would have multiple child views each can have multiple fragments). :
<h1>Hello World</h1>
<div id="template">
     <fragment src="http://localhost:3000"/> 
</div>
<div id="some-component">{{somevalue}}</div>

The "somevalue" is being populated by Handlebars during server rendering. In case if we don't use Handlebars what should be the best way to assign values from node object to views. E.g. I want my host and port no. for different env coming from some config file. Any pointers should help.

**Comment by stevoPerisic** [comment]
You can use handlebars as long as you process the template string before you return it from Tailor. What I've done is create a custom fetchTemplate function. In that function you will have access to the template HTML string which you can feed into the handlebars compile function to get your data into the template.
Let me know if you have any more questions.
